### ISSUE TITLE
2 Devices Configuration Home Assistant

### ISSUE THREAD

**Issue by alexmaria8** [wmbusmeters/wmbusmeters#1076]
### Type of request

Question

### HAOS version

2023.9.2

### wmbusmeters version

1.14.0

### Your message goes here

Hello!
Running the wmbusmeters addon on home assistant already with 1 device.
I got another im871a-USB device because i want to run 2 of them at the same time. The first one is working fine,
but I fail at the confifuration respectively it won't save 2 device after a restart of the wmbusmeter-addon neither a system-restart of Home Assistant.

What i tried:
**1) At the wmbusmeters HA addon configuration i clicked on 'Add a new parameter' named it: device = /dev/ttyUSB3:im871a:s1
The first one i got in use is already named: device = /dev/ttyUSB2:im871a:c1:t1
After i clicked on 'Save Changes' the addon restarts but than theres only 1 device (the newly added) and the other one is gone.**

**2) I edited the wmbusmeter.conf file from this:**
_device=/dev/ttyUSB2:im871a:c1,t1
donotprobe=/dev/ttyAMA0
format=json
logfile=/dev/stdout
loglevel=normal
logtelegrams=false
shell=/wmbusmeters/mosquitto_pub.sh "wmbusmeters/$METER_NAME" "$METER_JSON"_
**to this:**
_device=/dev/ttyUSB2:im871a:c1,t1
device=/dev/ttyUSB3:im871a:s1
donotprobe=/dev/ttyAMA0
format=json
logfile=/dev/stdout
loglevel=normal
logtelegrams=false
shell=/wmbusmeters/mosquitto_pub.sh "wmbusmeters/$METER_NAME" "$METER_JSON"_
**saved the file (i closed the file and reopnened it to check) but after a full restart of the system it looked again like before (only 1 device):**
_device=/dev/ttyUSB2:im871a:c1,t1
donotprobe=/dev/ttyAMA0
format=json
logfile=/dev/stdout
loglevel=normal
logtelegrams=false
shell=/wmbusmeters/mosquitto_pub.sh "wmbusmeters/$METER_NAME" "$METER_JSON"_

Can you plz help me?

**Comment by alexmaria8** [comment]
> Try to use the identifiers of your im871a-USB's to separate them
> 
> * the ID is the last 8 digits of the serial number printed on the stick
> 
> device=/dev/ttyUSB0:im871a[00102759]:t1
> 
> wmbusmeters log shows you the IDs:
> 
> Started config im871a[00102759] on /dev/ttyUSB0 listening on t1

Hey lindsand!
Thank you for your reply! Sadly it gives me the same results as before, so only 1 device stays the other one not.
Here are some screenshots (First variant I tried to change it in the wmbusmeter configuration UI and 2nd try to change the wmbusmeter.conf file).

Variant 1 (1st pic is initial situation, 2nd what i added and 3rd the result after saving and restarting the addon):
![1](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/aa528ced-c0fc-4c39-a846-17fa7c5a90d5)
![2](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/fbb5ad90-235c-4251-b52a-31af641febd2)
![3](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/9746d7db-b60d-42a4-9f26-4359a8a7e61e)
![4](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/0387c228-1fb8-432c-8103-35b24afc22e0)

Variant 2 (1st pic is initial situation, 2nd what i added and saved and 3rd the result after a restart of the system):
![A](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/4b3ed71a-a9f6-4d5c-a76d-3f18b923ee58)
![B](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/890d823c-13cd-4aba-8acd-0b3751ecea55)
![C](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/37422d8f-5c96-4f10-8486-50a1a2e45b84)

**Comment by weetmuts** [comment]
Perhaps HA cannot handle multiple identical parameters? @BIBOLV

**Comment by lindsand** [comment]
Put them on the same line separated with ;

>>wmbusmeters configuration
>>Content of this section will be used as wmbusmeters.conf. If two values for configuration parameter is needed, like:

>>donotprobe: /dev/ttyAMA0
>>donotprobe: /dev/ttyUSB0
>>**In web configuration interface it should be provided in one line** using ; delimiter between values, like: >>donotprobe=/dev/ttyAMA0;/dev/ttyUSB0

I made a mistake in the configuration some days ago and set up a logfile that didn't exist preventing wmbusmeters from starting. I tried to correct it by editing wmbusmeters.conf but it didn't help, the same wrong value appeared in wmbusmeters.conf after each restart. So I came to the conclusion that the UI-content is stored somewhere else and wmbusmeters.conf is overwritten with UI-data after each restart. So I had to uninstall wmbusmeters totally and start from scratch.

**Comment by BIBOLV** [comment]
Yes, two parameters can be provided in one line using ; as delimiter.
Yes, manually edited config file will be reseted on each restart. Config is stored in HA internal db and written to config on each restart.
@lindsand - you could have reseted config without reinstall trough addon configuration tab.

**Comment by lindsand** [comment]
>>you could have reseted config without reinstall trough addon configuration tab

No, wmbusmeter kept on crashing before addon configuration tab became available. So I was stuck.

**Comment by alexmaria8** [comment]
@lindsand @BIBOLV @weetmuts

Thank u all for ur help! **Using the delimiter  ;  was the key.** Now i can go on.

Just for people who encounter the same issue (in the future) i post the screenshot with the working HA addon config below. (I checked the wmbusmeters.conf later too and it keeps the second device. Getting it from the UI configuration as u already mentioned)
THX again!

![1](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/cc124ca0-fdf8-4761-9e8d-015404a6dde0)
![2](https://github.com/wmbusmeters/wmbusmeters/assets/110693814/ae43b12b-a2a5-4889-a45e-8e46534e8e86)

**Comment by BIBOLV** [comment]
@lindsand - you are talking about config tab at sidebar or one within HA addon related configuration alongside documentation tab?

**Comment by lindsand** [comment]
Config tab at sidebar, I made a mistake in the logfile name:

![image](https://github.com/wmbusmeters/wmbusmeters/assets/66431079/90bd14e7-25ed-4ec8-b7ba-fac082e7dfb5)

End result was that wmbusmeters crashed at startup (because it couldn't write to the log) and again after each restart, watchdog was on. So it was looping. And editing wmbusmeters.conf didn't change anything, old values popped back.

I noticed this possibility only afterwards:
![image](https://github.com/wmbusmeters/wmbusmeters/assets/66431079/75a65419-20ec-4941-9b61-d7dac12e6674)

I wanted to log some meter readings and after some trials I found out it can be done by adding a line in  wmbusmeters configuration:

meterfiles=/config/wmbusmeters/logs/meter_readings
